tests/functional_tests.rs

```diff
@@ -14,6 +14,10 @@ mod tests {
 
         // FIXME: As of right now, this just fails on rustc compilation which is not what
         // we want to check
-        assert!(Harness::check_folder("invalid_code", FileType::Bin).is_err())
+        assert!(Harness::check_folder("invalid_code", FileType::Bin).is_err());
+
+        std::env::set_var("GCCRS_EXTRA_ARGS", "-Werror");
```

Comment:
This will only work fine if you ensure that no other test is running at the same time. Currently there are no other tests, but  I don't know if that will change in the future.

Reply:
Yeah, but we are having issues with that in the whole harness since we also use `env::set_current_dir()`. I think it should only be "one test" that calls multiple functions. I mentioned it somewhere in a previous PR about testing, that since there was no way to set `--test-threads=1` from a configuration file it was impossible to do otherwise